### DISCUSSION TITLE
Is it possible to use Expose with a project that is not write in PHP ?

### DISCUSSION THREAD

**Discussion by PAGpK** [exposedev/expose#319] [Q&A]
Hello everybody,

I was try to use Expose with my Laravel project and it's work perfectly !

So i was try with my VueJS and Django project. I was very disappointed because Expose don't return https reponse (picture below).

Expose only works with PHP projects ? Or there is a specific configuration for the other projects ?

![Capture d’écran 2022-04-08 120752](https://user-images.githubusercontent.com/96131686/162416059-05392fc9-60d6-48b6-88d8-c15b4c52df17.png)

**Comment by MkLHX** [comment]
i can say yes!
It's not simple to get the auth_token when you use your own server instance.

Because all the front GUI like expose.dev doesn't exist in the project. 

So you cannot create an account and get your own token from the server dashboard. (dont hesitate to correct me but after many hours to read the source code i just fin da template about login but nothing else.)

So i manage to get my own account by using /api/users endpoint. I made a curl request in pure php script using the server basic auth header (i know username:password because it's my own server instance)
when i request the api i check if user exist i take the auth_token. if not i create the user and get the auth_token after.

Then i can run my Expose's agent on client side with my auth_token on my docker container.
